packages/container/lib/container.js

```diff
@@ -656,6 +664,21 @@ class FactoryManager {
       injectDeprecatedContainer(prototype, this.container);
     }
 
+    // required to allow access to things like
+    // the customized toString, _debugContainerKey,
+    // owner, etc. without a double extend and without
+    // modifying the objects properties
+    if (this.class._initFactory) {
```

Comment:
`typeof this.class._initFactory === 'function'` (more correct, and makes the JIT happy)